src/MongoDB.Driver.Core/Core/Authentication/AwsSignatureVersion4.cs

```diff
@@ -0,0 +1,203 @@
+﻿/* Copyright 2020–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace MongoDB.Driver.Core.Authentication
+{
+    /// <summary>
+    /// The AWS signature version 4.
+    /// </summary>
+    public class AwsSignatureVersion4
+    {
+        /// <summary>
+        /// Gets canonical headers.
+        /// </summary>
+        /// <param name="requestHeaders">The request headers.</param>
+        /// <returns>The canonical headers.</returns>
+        public static string GetCanonicalHeaders(SortedDictionary<string, string> requestHeaders)
```

Comment:
From my point of view this method improves readability, so I would like to leave it as is, unless there are other concerns.

Reply:
> a static class to have a ToString() method?

My point was that I don't see how changing a form from dictionary to a string can be named getting canonical headers? It sounds like just technical operation. In other words what was done in the scope of this method to call it getting canonical headers? Maybe I'm missing something. It's a minor point and I can rely on other reviewers here

Reply:
I checked this link https://raw.githubusercontent.com/mongodb/specifications/master/source/auth/includes/calculating_a_signature.png and I see where this came from.
However, it's still confusing me a bit, since the dictionary before this method also was in canonical form, the only change here that we change a format of this data